clarity/src/vm/functions/sequences.rs

```diff
@@ -383,3 +383,88 @@ pub fn special_slice(
     }?;
     Ok(sliced_seq)
 }
+
+pub fn special_replace_at(
+    args: &[SymbolicExpression],
+    env: &mut Environment,
+    context: &LocalContext,
+) -> Result<Value> {
+    check_argument_count(3, args)?;
+
+    let seq = eval(&args[0], env, context)?;
+    let seq_type = TypeSignature::type_of(&seq);
+    let (expected_elem_type, seq_is_list_type) =
+        if let TypeSignature::SequenceType(seq_subtype) = seq_type.clone() {
+            (seq_subtype.unit_type(), seq_subtype.is_list_type())
+        } else {
+            return Err(CheckErrors::ExpectedSequence(seq_type).into());
+        };
+    let index_val = eval(&args[1], env, context)?;
+    let new_element = eval(&args[2], env, context)?;
+
+    if expected_elem_type != TypeSignature::NoType && !expected_elem_type.admits(&new_element) {
+        return Err(CheckErrors::TypeValueError(expected_elem_type, new_element).into());
+    }
+
+    let index = if let Value::UInt(index) = index_val {
+        index
+    } else {
+        return Err(CheckErrors::TypeValueError(TypeSignature::UIntType, index_val).into());
+    };
+
+    match seq {
+        // todo: update the ClarityCostFunction once the Clarity2 related cost functions are implemented.
+        Value::Sequence(seq) => {
+            let seq_len = seq.len();
+            runtime_cost(ClarityCostFunction::Unimplemented, env, seq_len as u64)?;
```

Comment:
Why would the runtime cost be `O(n)` for an n-item sequence?  Can't this be `O(1)`?

Reply:
I mirrored the cost of the concat function - this function also has to perform an append.

Reply:
`concat` performs `O(n)` work because it has to copy data.  This does not.

Reply:
Is this supposed to be in-place or return a new seq?

Reply:
I'd strongly prefer that this method implement an in-place item replacement, because doing so would be much more efficient (and would save multiple heap allocations).  I misspoke earlier -- I meant to say that this code should not need to do a full sequence copy.